src/nvim/screen.c

```diff
@@ -299,6 +290,19 @@ void redraw_buf_status_later(buf_T *buf)
   }
 }
 
+/// called when the window bars for the buffer 'buf' need to be updated
+void redraw_buf_winbar_later(buf_T *buf)
```

Comment:
let's just do it in `redraw_buf_status_later` . we can add specific functions once we need them, not in advance.